internal/pkg/earpointer/analysis.go

```diff
@@ -65,33 +67,61 @@ func run(pass *analysis.Pass) (interface{}, error) {
 // Insert into the state all the local references and global references.
 func (vis *visitor) initReferences(ssainput *buildssa.SSA) {
 	state := vis.state
+	prog := ssainput.Pkg.Prog
 	for _, member := range ssainput.Pkg.Members {
-		if m, ok := member.(*ssa.Global); ok {
+		switch m := member.(type) {
+		case *ssa.Global:
 			if typeMayShareObject(m.Type()) &&
 				!strings.HasPrefix(m.Name(), "init$") { // skip some synthetic variables
 				state.Insert(MakeGlobal(m))
 			}
+		case *ssa.Function:
+			vis.initFunction(m)
+		case *ssa.Type:
+			for _, meth := range typeutil.IntuitiveMethodSet(m.Type(), &prog.MethodSets) {
+				if mv := prog.MethodValue(meth); mv != nil {
+					vis.initFunction(mv)
+				}
```

Comment:
We're a bit out of depth here, but is there any concern about double-initializing something that embeds?  I'm not sure how that method set resolves in, e.g.,

```golang
type Bar struct {}
func (Bar) Method() {}

type Foo struct {
  Bar
}
```
Will both `Foo.Method()` and `Bar.Method()` be produced by `IntuitiveMethodSet` / produce the same `*ssa.Function` from `prog.MethodValue`?  Is it a problem in `initFunction` if they do?  I see that you have handling in `initFunction` explicitly for anonymous fields, so I'm guessing you've already considered this.

Would be a good test-case to include, in any event.

Reply:
Good point! I added a test.
In general this is not a real issue because (1) SSA will produce Method() for Foo and Bar individually; so Method() won't be reused. (2) "state.Insert(ref)" will insert "ref" into the state only when ref is not already in the state. 
A more serious issue is that when a reference is not in a state while it is used. So here we want to instantiate all the local/global variables.